### PR TITLE
fix(core): update eslintrc config to show no-restricted-imports rules

### PR DESCRIPTION
**Pull request by pedrobonamin**
### Description
The existing eslint config is overriding the original `no-restricted-imports` rule.
When this rule was added:
```
'no-restricted-imports': [
          'error',
          {
            paths: [
              {
                name: 'react',
                importNames: ['createContext'],
                message: 'Please place context in _singletons',
              },
            ],
          },
```

The previous rules stopped working.

- This PR introduces changes to make all the `no-restricted-imports` rules work at the same time.
- This PR also fixes the failing checks, by either importing the corresponding files or opting out where needed.

With the changes introduced, now we are seeing all the expected errors.
Screenshot 2024-08-14 at 09 55 18

### What to review

- Are this changes ok? 
- Is there any way to configure the eslint file to expose different error messages for React default import error and context error ?
- **Small caveat:** The error for "Please place context in _singletons" is not standalone anymore, it's shared with the React default error. 

### Testing

Create a .tsx file inside `packages/sanity/*` and add the following code, the errors should show in the file and when running the lint check action.
```
import {Button} from '@sanity/ui'
import React, {createContext} from 'react'
import styled from 'styled-components'

const StyledButton = styled(Button)`
  padding: 10px;
`
const context = createContext(null)

export function Test() {
  const [count, setCount] = React.useState(0)
  return <StyledButton onClick={() => setCount(count + 1)} text={`count: ${count}`} />
}
```

### CODE REVIEW

File: packages/sanity/src/core/form/studio/tree-editing/components/search/TreeEditingSearch.tsx [hunk 1]
```diff
@@ -138,7 +139,6 @@ export function TreeEditingSearch(props: TreeEditingSearchProps): JSX.Element {
 
   return (
     <StyledPopover
-      animate
```
Comment (pedrobonamin) [inline packages/sanity/src/core/form/studio/tree-editing/components/search/TreeEditingSearch.tsx:141]: This is the default in the `ui-components` popover

File: packages/sanity/src/core/form/studio/tree-editing/components/TreeEditingDialog.tsx [hunk 2]
```diff
@@ -219,6 +225,7 @@ export function TreeEditingDialog(props: TreeEditingDialogProps): JSX.Element |
           <Card borderTop>
             <Flex align="center" justify="flex-end" paddingX={3} paddingY={2} sizing="border">
               <Button
+                size="large"
```
Comment (pedrobonamin) [inline packages/sanity/src/core/form/studio/tree-editing/components/TreeEditingDialog.tsx:228]: The large `ui-components` button equals to the default size of the button imported from `@sanity/ui`

**Review by RitaDias — APPROVED** [review]
Tested and double checked with the modal and everything looks good :) 
The only thing I don't know _a whole lot_ about is the setting up of the linting but it looks good to me as well!

**Review by binoy14 — APPROVED** [review]
Good catch! thanks for fixing it

**Pull request merged**